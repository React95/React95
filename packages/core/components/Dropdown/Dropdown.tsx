import React from 'react';
import styled from '@xstyled/styled-components';
import { th } from '@xstyled/system';

import caret from '../shared/assets/pattern/downcaret.png';

const Wrapper = styled.div`
  position: relative;
  width: 200px;
  height: 20px;

  &:after {
    display: flex;
    justify-content: center;
    width: 17px;
    height: 17px;
    font-size: 14px;
    line-height: 1.1;
    content: '';
    pointer-events: none;
    position: absolute;
    right: 1px;
    top: 2px;

    background-color: bg;
    box-shadow: inset 0.5px 0.7px 0px 0.7px ${th('colors.grays.2')},
      inset -1px 0px 0 1px ${th(
        'colors.grays.3',
      )}, inset 1.5px 1.5px 0px 1.5px ${th('colors.white')};

    border-right: 1;
    border-bottom: 1;

    background-image: url('${caret}');
    background-position: center center;
    background-repeat: no-repeat;
  }
`;

const Select = styled.select`
  position: relative;
  outline: none;
  border: none;
  border-radius: 0;
  width: 100%;
  height: 20px;

  padding: 3;

  background-color: white;

  border-left: 1;
  border-left-color: grays.3;
  border-top: 1;
  border-top-color: grays.3;

  box-shadow: inset -1px -1px 0 0 ${th('colors.grays.2')},
    inset 1px 1px 0 0 ${th('colors.black')},
    0.5px 0.5px 0 0.5px ${th('colors.white')};

  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
`;

type DropdownProps = {
  options?: Array<string | number>;
} & React.HTMLAttributes<HTMLSelectElement>;

const Dropdown: React.FC<DropdownProps> = ({ options, ...rest }) => (
  <Wrapper>
    <Select {...rest}>
      {options &&
        options.map(option => (
          <option key={option} value={option}>
            {option}
          </option>
        ))}
    </Select>
  </Wrapper>
);

Dropdown.defaultProps = {
  options: [
    '',
    'C:\\Documents and Settings',
    'C:\\Documents and Settings\\Documents',
    'iexplorer.exe',
  ],
};

export default Dropdown;
