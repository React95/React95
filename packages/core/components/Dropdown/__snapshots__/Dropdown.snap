// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Dropdown /> Snapshots should match snapshot with options prop 1`] = `
.c0 {
  position: relative;
  width: 200px;
  height: 20px;
}

.c0:after {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 17px;
  height: 17px;
  font-size: 14px;
  line-height: 1.1;
  content: '';
  pointer-events: none;
  position: absolute;
  right: 1px;
  top: 2px;
  background-color: #c3c7cb;
  box-shadow: inset 0.5px 0.7px 0px 0.7px #c3c7cb, inset -1px 0px 0 1px #868a8e,inset 1.5px 1.5px 0px 1.5px #ffffff;
  border-right: 1px solid;
  border-bottom: 1px solid;
  background-image: url('test-file-stub');
  background-position: center center;
  background-repeat: no-repeat;
}

.c1 {
  position: relative;
  outline: none;
  border: none;
  border-radius: 0;
  width: 100%;
  height: 20px;
  padding: 3px;
  background-color: #ffffff;
  border-left: 1px solid;
  border-left-color: #868a8e;
  border-top: 1px solid;
  border-top-color: #868a8e;
  box-shadow: inset -1px -1px 0 0 #c3c7cb, inset 1px 1px 0 0 #000000, 0.5px 0.5px 0 0.5px #ffffff;
  -webkit-appearance: none;
  -moz-appearance: none;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
}

<div>
  <div
    class="c0"
  >
    <select
      class="c1"
    >
      <option
        value=""
      />
      <option
        value="C:\\\\Windows"
      >
        C:\\Windows
      </option>
      <option
        value="C:\\\\Windows/System32"
      >
        C:\\Windows/System32
      </option>
    </select>
  </div>
</div>
`;
